csrc/python_frontend/fusion_definition.h

```diff
@@ -240,6 +250,17 @@ class NVF_API FusionDefinition : public FusionState {
 
  private:
   mutable std::optional<std::string> debug_output_ = std::nullopt;
+
+#ifdef NVFUSER_DISTRIBUTED
+  //! DO NOT COMMIT TO THESE CHANGES!
+  //! The reason we have these is due to the lack of cache for multidevice
+  //! executor
+  bool multidevice_flag = false;
```

Comment:
I smelled redundancy between multidevice_flag_ and multi_device_executor_. I understand they are set at different times. However, before any fusion is executed, `multi_device_executor_` is not null iff `multidevice_flag` is true. Correct? 

If so, we may be able to set multi_device_executor directly in setMultiDevice instead of using multidevice_flag_ as a proxy.

Reply:
That's a good call! Let me refactor it.

Reply:
Need to have this one first. #2051

Reply:
For now, `MultiDeviceExecutor` constructor takes a `Fusion` as an argument (similarily to `FusionExecutorCache` and `FusionExecutor`) so it is not possible to instantiate it before the fusion is defined